### PR TITLE
Use HTTPS if location protocol is HTTPS in iD.Connection

### PR DESCRIPTION
**Pull request by frewsxcv**
Relevant to https://github.com/strava/iD/issues/5#issuecomment-109800372

### CODE REVIEW

**Comment by pnorman** [comment]
Looks like the tests need revising too.

**Comment by frewsxcv** [comment]
I fixed all the tests except one. If anyone has ideas about what is going wrong there, let me know

**Comment by pnorman** [comment]
```
  iD.modes.AddPoint
    clicking the map
XMLHttpRequest cannot load file://www.openstreetmap.org/api/0.6/map?bbox=-77.02514648437499,38.89958342598271,-77.01965332031249,38.90385833966776. Cross origin requests are only supported for HTTP.
  1) "before each" hook
```

Ugh. This is a nasty problem.

You can't assume that the OSM API is accessible on the same protocol that iD is loaded by, because iD could be loaded locally over file://

**Comment by jfirebaugh** [comment]
Chrome has some restrictions for resources loaded via `file://` that generally make loading it that way unadvisable. That said, it's probably better to do something like `location.protocol === 'http' ? 'http' : 'https'`.

**Comment by frewsxcv** [comment]
How does it look now? I added an optional parameter to the `iD.Connection` constructor mainly because I didn't want to figure out how to mock `window.location` in order to write a regression test

**Comment by frewsxcv** [comment]
Alternatively, we could just make all connections secure and disallow HTTP...

**Comment by frewsxcv** [comment]
Anyone have thoughts on this?

**Comment by pnorman** [comment]
Not all API instances allow HTTPS.

**Comment by frewsxcv** [comment]
My current changes do not force HTTPS, unless the user is already on an HTTPS site

**Comment by pnorman** [comment]
> My current changes do not force HTTPS, unless the user is already on an HTTPS site

Yes, but you were suggesting disallowing HTTP

**Comment by frewsxcv** [comment]
> Yes, but you were suggesting disallowing HTTP

Yeah, I understand. Sorry I didn't make that clear in my previous message.

**Comment by bhousel** [comment]
I notice that using the [source switcher](https://github.com/openstreetmap/iD/blob/master/index.html#L261-L272) on an https site will break this, but it's broken anyway because `api06.dev.openstreetmap.org` doesn't support https, so I guess not that big a deal.

Thanks for adding this, @frewsxcv

File: js/id/core/connection.js [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
-        url = 'http://www.openstreetmap.org',
+        protocol = useHttps ? 'https:' : 'http:',
+        url = protocol + '//www.openstreetmap.org',
         connection = {},
         inflight = {},
         loadedTiles = {},
         tileZoom = 16,
         oauth = osmAuth({
-            url: 'http://www.openstreetmap.org',
+            url: protocol + '//www.openstreetmap.org',
```
Comment (1ec5) [inline js/id/core/connection.js]: This is unnecessary: just start the URL with `//`, and the current protocol will be used (except in Netscape Navigator 4, I suppose).

Comment (frewsxcv) [inline js/id/core/connection.js]: Look through he discussion the associated pull request. Sometimes, we are retrieving this js file via the `file://` protocol which results in an undesired outcome

Comment (1ec5) [inline js/id/core/connection.js]: Doh! Sorry about that. :smile:

**Pull request merged**